### ISSUE TITLE
DIfferent `:base_url` for different sessions?

### ISSUE THREAD

**Issue by peaceful-james** [elixir-wallaby/wallaby#570]
Is it possible to use different `:base_url` for different sessions?

**Comment by mhanberg** [comment]
Currently, no, but that could be a new feature.

What is your use case?

**Comment by peaceful-james** [comment]
> Currently, no, but that could be a new feature.
> 
> What is your use case?

My use case is a monorepo with two projects - an Elixir/Phoenix API and LiveView site running on localhost:4000 and a React SPA running on localhost:3000. I want to test full "user flows" where customers use the SPA on 3000 and staff/admin use the LiveView site on 4000. These two apps use Graphql Subscriptions and LiveView so an action by a customer may make staff view update (with no interaction).

Basically, i want to test full user stories spanning multiple realtime apps.

**Comment by mhanberg** [comment]
Gotcha, that makes sense.

Would something like this work for you?

```elixir
defmodule MyAppWeb.TwoAppFeatureTest do
  use ExUnit.Case
  use Wallaby.Feature

  @sessions [[base_url: "http://localhost:3000"], [base_url: "http://localhost:4000"]]
  feature "stuff happens", %{sessions: [spa, admin]} do
    spa |> login_user
    admin |> login_admin

    spa
    |> visit("/some-path")
    |> click(Query.button("Do Thing"))

    admin
    |> assert_text("some indicator that the user did a thing")
  end
end
```

**Comment by michallepicki** [comment]
Hey @mhanberg @peaceful-james , until there's a nicer API, I know there are work-arounds. I think I had a test where I needed something like that, and I leveraged the fact that it is possible to use [Browser.visit/2](https://hexdocs.pm/wallaby/Wallaby.Browser.html#visit/2) with absolute paths. I had the `:base_url` set to an empty string, and to make it easier to visit appropriate base URL depending on context, I had helper functions. Something like `spa |> visit(spa_path("/some_path"))`. Maybe you knew that already, but maybe this note will help someone else :)

**Comment by peaceful-james** [comment]
Thanks for the information, both of you. The application I wanted to do this for has been finished for months now and I am knee-deep in a new project.
If I ever revisit this, i will be sure to try both of your techniques (although I think i have actually tried both of them). I always found that the main problem seems to be the length/duration of Wallaby tests. Small/short tests would pass but anything with extended bouts of interaction would randomly fail.
